.version_information.sh

```diff
@@ -24,8 +24,8 @@ ONNX_RUNTIME_VERSION=$(python -c "exec(\"try:\n  import onnxruntime; print(onnxr
 
 CMAKE_VERSION=$(cmake --version  | grep version | awk '{print $3}')
 if [[ $(command -v nvidia-smi) ]]; then
-  GPU_INFOS=$(nvidia-smi --query-gpu=name,memory.total,driver_version --format=csv | tail -n -1)
-  NUM_GPUS=$(echo $GPU_INFOS | wc -l)
+  GPU_INFOS=$(nvidia-smi --query-gpu=name,memory.total,driver_version --format=csv,noheader)
+  NUM_GPUS=$(nvidia-smi --query-gpu=name --format=csv,noheader | wc -l)
```

Comment:
```suggestion
  NUM_GPUS=$(echo $GPU_INFOS | wc -l)
```

I'd suggest to keep the current implementation here to potentially save a millisecond or two ;)

Reply:
`echo` collapses `$GPU_INFOS` into a single line, causing `wc -l` to return `1` even if multiple GPUs exist.

I would recommend to keep `NUM_GPUS=$(nvidia-smi --query-gpu=name --format=csv,noheader | wc -l)`

Alternatively, one could use `NUM_GPUS=$(echo "$GPU_INFOS" | wc -l)` to preserve the line breaks (by adding `"` around `$GPU_INFOS`). This solution is not as easy to understand though, as it requires knowledge of how `echo` handles line breaks.

Reply:
I'd use the echo method. The entire script is based on non-readable grepping, seding, awking, etc. The intention of the line is clear.

Reply:
Ok, I changed the line to `NUM_GPUS=$(echo "$GPU_INFOS" | wc -l)`.